examples/compiler_selection/Older.sol

```diff
@@ -0,0 +1,12 @@
+//SPDX-License-Identifier: UNLICENSED
+pragma solidity >=0.7.0 <0.8.0;
+
+// Underflow is illegal starting in Solidity 0.8.0
+// https://blog.soliditylang.org/2020/12/16/solidity-v0.8.0-release-announcement/
+// Therefore this code would fail to compile unless the `solidity_version` attribute is honored.
```

Comment:
It would still compile, but would "revert" at runtime (effectively a hybrid between throwing an exception and rolling back a transaction). So the `pragma` is what's exercising the version check here.

Reply:
Well, the 0.8.* compiler will throw an error if it processes this file, is what I meant.

Reply:
okay then I'm just being overly descriptive of what is the use case for the version pinning. You're right that you'd want identical contracts anyway, so it's not strictly needed.